x-pack/plugins/uptime/public/components/settings/default_email.tsx

```diff
@@ -0,0 +1,157 @@
+import { connectorsSelector } from '../../state/alerts/alerts';
+import { DefaultEmail as DefaultEMailType } from '../../../common/runtime_types';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { UptimePluginServices } from '../../apps/plugin';
+
+export function DefaultEmail({
+  value,
+  isLoading,
+  isDisabled,
+  onChange,
+  connectors,
+}: {
+  value?: DefaultEMailType;
+  isLoading: boolean;
+  isDisabled: boolean;
+  onChange: OnFieldChangeType;
+  connectors?: string[];
+}) {
+  const [toEmail, setToEmail] = useState<string[]>(value?.to ?? []);
+  const [ccEmail, setCcEmail] = useState<string[]>(value?.cc ?? []);
+  const [bccEmail, setBccEmail] = useState<string[]>(value?.bcc ?? []);
+
+  const [errors, setErrors] = useState({});
+
+  const { actionTypeRegistry } = useKibana<UptimePluginServices>().services.triggersActionsUi;
+
+  const { data = [] } = useSelector(connectorsSelector);
+
+  useEffect(() => {
+    const inValidEmail = toEmail.find((val) => !validateEmail(val));
+    const inValidCcEmail = ccEmail.find((val) => !validateEmail(val));
+    const inValidBccEmail = bccEmail.find((val) => !validateEmail(val));
+
+    const emailError = toEmail.length === 0 ? REQUIRED_EMAIL : undefined;
+
+    setErrors({
+      to: getInvalidEmailError(inValidEmail) || emailError,
+      cc: getInvalidEmailError(inValidCcEmail),
+      bcc: getInvalidEmailError(inValidBccEmail),
+    });
+
+    onChange({
+      defaultEmail: {
+        to: toEmail,
+        cc: ccEmail,
+        bcc: bccEmail,
+      },
+    });
+    // we don't need onChange in dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [ccEmail, toEmail, bccEmail]);
+
+  useEffect(() => {
+    setToEmail(value?.to ?? []);
+    setCcEmail(value?.cc ?? []);
+    setBccEmail(value?.bcc ?? []);
+  }, [value]);
+
+  if (
+    !data?.find(
+      (connector) => connectors?.includes(connector.id) && connector.actionTypeId === '.email'
+    )
+  ) {
+    return null;
+  }
+
+  const emailActionType = actionTypeRegistry.get('.email');
+  const ActionParams = emailActionType.actionParamsFields;
+
+  const onEmailChange = (key: string, val: string[]) => {
+    switch (key) {
+      case 'to':
+        setToEmail(val);
+        break;
+      case 'cc':
+        setCcEmail(val);
+        break;
+      case 'bcc':
+        setBccEmail(val);
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <EuiDescribedFormGroup
+      title={
+        <h4>
+          <FormattedMessage
+            id="xpack.uptime.sourceConfiguration.alertConnectors.defaultEmail"
+            defaultMessage="Default email"
+          />
+        </h4>
+      }
+      description={
+        <FormattedMessage
+          id="xpack.uptime.sourceConfiguration.defaultConnectors.description.defaultEmail"
+          defaultMessage="Email settings required for selected email alert connectors."
```

Comment:
Might need to adjust this content. Should connectors be plural? Isn't there just one email connector?